### PR TITLE
feat: add extract vk command in maci-cli

### PR DESCRIPTION
**Pull request by kittybest**
# Description

Add `extractVk` command in maci-cli package.

## Confirmation

- [x] I have read and understand MACI's [contributor guidelines](https://maci.pse.dev/docs/contributing) and [code of conduct](https://maci.pse.dev/docs/contributing/code-of-conduct).
- [x] I have read and understand MACI's [GitHub processes](https://github.com/privacy-scaling-explorations/maci/discussions/847).
- [x] I have read and understand MACI's [testing guide](https://maci.pse.dev/docs/testing).

### CODE REVIEW

File: cli/ts/index.ts [hunk 1]
```diff
@@ -357,6 +358,37 @@ program
...
+  .requiredOption(
+    "-tnqv, --tally-votes-zkey-non-qv <tallyVotesZkeyPathNonQv>",
+    "the tally votes non-qv zkey path (see different options for zkey files to use specific circuits https://maci.pse.dev/docs/trusted-setup, https://maci.pse.dev/docs/testing/#pre-compiled-artifacts-for-testing)",
+  )
+  .action(async (cmdObj) => {
+    try {
+      const processVkQv = await extractVk(cmdObj.processMessagesZkeyQv);
+      const tallyVkQv = await extractVk(cmdObj.tallyVotesZkeyQv);
+      const processVkNonQv = await extractVk(cmdObj.processMessagesZkeyNonQv);
+      const tallyVkNonQv = await extractVk(cmdObj.tallyVotesZkeyNonQv);
```
Comment (0xmad) [inline cli/ts/index.ts:385]: These can be done with Promise.all

File: cli/ts/index.ts [hunk 2]
```diff
@@ -357,6 +358,37 @@ program
...
+
+      fs.writeFileSync("vks.json", JSON.stringify({ processVkQv, tallyVkQv, processVkNonQv, tallyVkNonQv }));
```
Comment (0xmad) [inline cli/ts/index.ts:387]: let's use promise version

File: cli/ts/index.ts [hunk 3]
```diff
@@ -357,6 +358,37 @@ program
...
+  )
+  .requiredOption(
+    "-pnqv, --process-messages-zkey-non-qv <processMessagesZkeyPathNonQv>",
+    "the process messages non-qv zkey path (see different options for zkey files to use specific circuits https://maci.pse.dev/docs/trusted-setup, https://maci.pse.dev/docs/testing/#pre-compiled-artifacts-for-testing)",
+  )
+  .requiredOption(
+    "-tnqv, --tally-votes-zkey-non-qv <tallyVotesZkeyPathNonQv>",
+    "the tally votes non-qv zkey path (see different options for zkey files to use specific circuits https://maci.pse.dev/docs/trusted-setup, https://maci.pse.dev/docs/testing/#pre-compiled-artifacts-for-testing)",
+  )
+  .action(async (cmdObj) => {
```
Comment (0xmad) [inline cli/ts/index.ts:380]: Action must be like other actions (move logic to commands folder)

File: cli/ts/index.ts [hunk 4]
```diff
@@ -357,6 +358,37 @@ program
       program.error((error as Error).message, { exitCode: 1 });
     }
   });
+program
+  .command("extractVk")
```
Comment (0xmad) [inline cli/ts/index.ts:362]: Let's add docs for this command

**Review by 0xmad — CHANGES_REQUESTED** [review]
@kittybest thanks, just left some comments

File: website/versioned_docs/version-v1.3_alpha/cli.md [hunk 5]
```diff
@@ -42,6 +42,7 @@ pnpm run hardhat
...
...                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               |
```
Comment (ctrlc03) [inline website/versioned_docs/version-v1.3_alpha/cli.md:45]: would change this to: "extract verification keys (vKey) from zKey files.

**Review by ctrlc03 — APPROVED** [review]
Thanks!

File: cli/ts/index.ts (hunk 2 above)
Comment (0xmad) [reply cli/ts/index.ts]: This is not fixed

Comment (ctrlc03) [reply cli/ts/index.ts]: oh you refer to fs.promises? my bad for marking this as resolved

Comment (kittybest) [reply cli/ts/index.ts]: I also don't get it OAO my bad

**Review by 0xmad — APPROVED** [review]
@kittybest thanks!

**Pull request merged**